File: C/src/cap1/hello1.c
/* 
Comandos:
 cd ~/Documentos/ProjetosUbuntu/C/
 nano src/prog1.c  (ou nano | prog1.c |  em caso de omissão, nome é solicitado com Ctrl+O)
 gcc src/prog1.c | -o bin/prog1 |   em caso de omissão, a.out é gerado automaticamente em ~/Documentos/ProjetosUbuntu/C/
 ./bin/prog1 argv1
 */ 
// #define SAUDACAO "OI, "  <- lembrança para depois 

//Programa: Manipulacao de Texto - Capitulo 1 (C)
//Ambiente: Ubuntu - nano, gcc

#include <stdio.h>

int main(int argc, char *argv[]){
    const char SAUDACAO[] = "Oi, ";
    char nome[10];

    printf("LITERAL CLASSICA HELLOWORLD:");
    printf("\n\x1B[7m%s\x1B[0m\n", "Oi, mundo!");
    
    printf("\n%s", "VALOR PADRAO DA VARIAVEL ANTES DA INICIALIZACAO:");
    printf("\n%s%s", "nome = ", nome);
    printf("\n%s%s", SAUDACAO, nome);
    
    sprintf(nome,"Ana");
    printf("\n\nINICIALIZACAO:\n");
    printf("nome = %s\n", nome);
    printf("%s%s\n", SAUDACAO, nome);
    
    sprintf(nome,"Bela");
    printf("\nNOVA ATRIBUICAO:\n");
    printf("nome = %s\n", nome);
    printf("%s%s\n", SAUDACAO, nome);
    
    printf("\nENTRADA INTERATIVA:\n");
    printf("Digite seu nome: ");
    scanf("%s", nome);
    printf("%s%s\n", SAUDACAO, nome);

    printf("\nENTRADA VIA ARGUMENTO:\n");
    printf("%s%s\n", SAUDACAO, argv[1]);
    
    printf("\nRESUMO:\n");
    printf("Constante:\tSAUDACAO = %s\n", SAUDACAO);
    printf("Variavel:\tnome = %s\n", nome);
    printf("Concatenacao:\t%s%s\n", SAUDACAO, nome);
    printf("\nMensagem com atribuicao da variavel\n");
    printf("nome:\t\t%s%s\n", SAUDACAO, nome);
    printf("argv[1]:\t%s%s\n", SAUDACAO, argv[1]);		
    	
    return 0;
}

File: Node/cap1/hello1.js
#!/usr/bin/env node
// Programa: helloworld, comentários e console.log
// Ambiente: Ubuntu - node.js, gedit 

/* Comentarios,
*  console.log,
*  literais string,
*  constantes e variaveis
/* asteriscos sao opcionais e
   comentarios aninhados precisam
   ser escapados assim: *\/
*/
// backslash escapa line break
// fim de linha com ; opcional

const SAUDACAO = "Oi\
, "                             // definicao obrigatoria

let nome                        /* declaracao obrigatoria */


console.log("LITERAL CLASSICA 'HELLOWORLD':")
console.log("\x1B[7m%s\x1B[0m", "Oi, mundo!")

console.log('\nVALOR PADRAO ANTES DA "INICIALIZACAO":')
console.log("%s%s", "nome = ", nome)
console.log("%s%s", SAUDACAO, nome)

nome = "Ana"
console.log('\nINICIALIZACAO ou PRIMEIRA \'ATRIBUICAO\':')
console.log("nome = ", nome)
console.log(SAUDACAO, nome)

nome = 'Beatriz'
console.log("\nNOVA \"ATRIBUICAO\":")
console.log("nome = " + nome)
console.log(SAUDACAO + nome)

console.log("\nRESUMO:")
console.log("Constante:\t" + "SAUDACAO = '" + SAUDACAO + "'")
console.log("Variavel:\t" + "nome = '" + nome + "'")
console.log("Concatenacao:\t'" + SAUDACAO + nome + "'")


/*
Avaliar
const SAUDACAO = `Oi,
`;

*/
